src/Duo.Clients.Wpf/Presentation/GestioneAnagraficaProdottiViewModel.cs

```diff
@@ -0,0 +1,90 @@
+﻿using Duo.Clients.Wpf.Services;
+using Duo.Domain.ViewModels.Prodotti;
+using Duo.Messages.Prodotti.Commands;
+using Radical.CQRS.Client;
+using System;
+using System.Collections.ObjectModel;
+using Topics.Radical.ComponentModel.Messaging;
+using Topics.Radical.Windows.Presentation;
+
+namespace Duo.Clients.Wpf.Presentation
+{
+    class GestioneAnagraficaProdottiViewModel : AbstractViewModel
+    {
+        readonly IMessageBroker broker;
+        readonly ProdottiViewsService prodottiViewsService;
+        readonly Services.AppSettings settings;
+
+
+        public GestioneAnagraficaProdottiViewModel(Services.AppSettings settings, IMessageBroker broker, Services.ProdottiViewsService prodottiViewsService)
+        {
+            this.settings = settings;
+            this.broker = broker;
+            this.prodottiViewsService = prodottiViewsService;
+
+            CaricaProdotti();
+
+            ManagePropertyMetadata();
+        }
+
+        private async void CaricaProdotti()
+        {
+            var view = await prodottiViewsService.List();
+            this.Prodotti = new ObservableCollection<ProdottoView>(view.Results);
+
+        }
+
+        private void ManagePropertyMetadata()
+        {
+            this.GetPropertyMetadata(() => this.ProdottoSelezionato)
+                .OnChanged(pvc =>
+                {
+                    this.ElementoSelezionato = (this.ProdottoSelezionato != null);
+                });
+        }
+
+        public ObservableCollection<ProdottoView> Prodotti
+        {
+            get { return this.GetPropertyValue(() => this.Prodotti); }
+            set { this.SetPropertyValue(() => this.Prodotti, value); }
+        }
+
+        public ProdottoView ProdottoSelezionato
+        {
+            get { return this.GetPropertyValue(() => this.ProdottoSelezionato); }
+            set { this.SetPropertyValue(() => this.ProdottoSelezionato, value); }
+        }
+
+        public bool ElementoSelezionato
+        {
+            get { return this.GetPropertyValue(() => this.ElementoSelezionato); }
+            set { this.SetPropertyValue(() => this.ElementoSelezionato, value); }
+        }
+
+        public void AggiungiProdotto()
+        {
+            this.broker.Broadcast(this, new Messaging.ApriManutenzioneProdottoMessage(Guid.Empty, 0, string.Empty, string.Empty,0));
```

Comment:
Mi autodenuncio… l’avevo visto e mi faceva schifo… e poi mi è passato di mente di sistemarlo. 

 

Devo sistemare lo stesso anche per i trattamenti. 

 

Lo faccio in questa PR o la mandiamo avanti e faccio un’altra dopo per questa e per i trattamenti?

 

Da: Mauro Servienti [mailto:notifications@github.com] 
Inviato: 31 March 2016 11:34
A: DuoMes/duo <duo@noreply.github.com>
Cc: Nazareno Manco <nazareno.m@live.com>
Oggetto: Re: [DuoMes/duo] Aggiunta gestione anagrafica prodotti (#60)

 

In src/Duo.Clients.Wpf/Presentation/GestioneAnagraficaProdottiViewModel.cs <https://github.com/DuoMes/duo/pull/60#discussion_r58025334> :

> +
> +        public ProdottoView ProdottoSelezionato
> +        {
> +            get { return this.GetPropertyValue(() => this.ProdottoSelezionato); }
> +            set { this.SetPropertyValue(() => this.ProdottoSelezionato, value); }
> +        }
> +
> +        public bool ElementoSelezionato
> +        {
> +            get { return this.GetPropertyValue(() => this.ElementoSelezionato); }
> +            set { this.SetPropertyValue(() => this.ElementoSelezionato, value); }
> +        }
> +
> +        public void AggiungiProdotto()
> +        {
> +            this.broker.Broadcast(this, new Messaging.ApriManutenzioneProdottoMessage(Guid.Empty, 0, string.Empty, string.Empty,0));

questa è teribbbile, riutilizzare lo stesso messaggio per due scopi diversi ti obbliga a ste brutture

—
You are receiving this because you were assigned.
Reply to this email directly or view it on GitHub <https://github.com/DuoMes/duo/pull/60/files/14b9e83407502201e28f51bc6b6dc3651c9b4a8c#r58025334>   <https://github.com/notifications/beacon/AKnparAqSBKjNQ7Tned-Yj2ivGLAdFeWks5py5UMgaJpZM4H7fBR.gif>